azurerm/resource_arm_container_group.go

```diff
@@ -197,6 +188,9 @@ func resourceArmContainerGroupRead(d *schema.ResourceData, meta interface{}) err
 	resp, err := containterGroupsClient.Get(resGroup, name)
 
 	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return nil
```

Comment:
given the Read method is used for several things (both checking that a resource exists at Plan time, and for parsing data from the API) - can we add `d.SetId("")` before the `return nil` here to ensure that deleted resources are detected?

Reply:
Done.